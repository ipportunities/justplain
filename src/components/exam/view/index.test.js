import React from 'react'
import { shallow } from 'enzyme'
import { ExamView } from './index'

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useSelector: jest.fn().mockReturnValue({ auth: { userType: 'coach' } })
}))

describe('Exam View', () => {
  it('should render correctly with coach view', () => {
    const component = shallow(<ExamView />)
    expect(component).toMatchSnapshot()
  })

  it('should render correctly with student view', () => {
    jest.mock('react-redux', () => ({
      ...jest.requireActual('react-redux'),
      useSelector: jest.fn().mockReturnValue({ auth: { userType: 'student' } })
    }))

    const component = shallow(<ExamView />)
    expect(component).toMatchSnapshot()
  })
})
