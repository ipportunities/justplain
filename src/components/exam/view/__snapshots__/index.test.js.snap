// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Exam View should render correctly with coach view 1`] = `
<div>
  <ExamViewStudent />
</div>
`;

exports[`Exam View should render correctly with student view 1`] = `
<div>
  <ExamViewStudent />
</div>
`;
